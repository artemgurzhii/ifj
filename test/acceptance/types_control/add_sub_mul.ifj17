Scope
dim a as integer
dim b as integer
dim c as integer

a = 1
b = 3

c = a + b
c = a - b
c = a * b
c = a / b

End Scope
