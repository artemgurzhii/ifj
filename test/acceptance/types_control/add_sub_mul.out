.IFJcode17
JUMP Scope
LABEL Scope
CREATEFRAME
DEFVAR TF@temp_bool_1
DEFVAR TF@temp_bool_2
DEFVAR GF@a
DEFVAR GF@b
DEFVAR GF@c
MOVE GF@a int@1
MOVE GF@b int@3
TYPE TF@temp_bool_1 GF@a
TYPE TF@temp_bool_2 GF@b
JUMPIFNEQ END_IF_0 TF@temp_bool_1 TF@temp_bool_2
ADD GF@c GF@a GF@b
TYPE TF@temp_bool_1 GF@a
TYPE TF@temp_bool_2 GF@b
JUMPIFNEQ END_IF_0 TF@temp_bool_1 TF@temp_bool_2
SUB GF@c GF@a GF@b
TYPE TF@temp_bool_1 GF@a
TYPE TF@temp_bool_2 GF@b
JUMPIFNEQ END_IF_0 TF@temp_bool_1 TF@temp_bool_2
MUL GF@c GF@a GF@b
DIV GF@c GF@a GF@b
LABEL END_IF_0
